### ISSUE TITLE
[2.0.0.2] Elasticache throws error when profile argument is specified

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29911]
From @streeter on 2016-02-01T04:03:16Z
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

elasticache module
##### ANSIBLE VERSION

2.0.0.2
##### SUMMARY

I got an error with boto 2.39.0 and ansible 2.0.0.2 when I specified the profile argument while creating a memcache cluster:

``` yml
    - name: Create memcache instance
      elasticache:
        name: memcache
        profile: "{{ ec2_profile }}"
        region: "{{ ec2_region }}"
        state: present
        engine: memcached
        cache_engine_version: 1.4.24
        node_type: cache.t2.micro
        num_nodes: 3
        cache_parameter_group: default.memcached1.4
        cache_port: 11211
        cache_subnet_group: memcache-subnet-group
```

Here's the error output:

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/Users/streeter/.ansible/tmp/ansible-tmp-1454299305.96-208803218223017/elasticache", line 2727, in <module>
    main()
  File "/Users/streeter/.ansible/tmp/ansible-tmp-1454299305.96-208803218223017/elasticache", line 512, in main
    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
  File "/Users/streeter/.ansible/tmp/ansible-tmp-1454299305.96-208803218223017/elasticache", line 2667, in get_aws_connection_info
    if not boto_supports_profile_name():
  File "/Users/streeter/.ansible/tmp/ansible-tmp-1454299305.96-208803218223017/elasticache", line 2583, in boto_supports_profile_name
    return hasattr(boto.ec2.EC2Connection, 'profile_name')
AttributeError: 'module' object has no attribute 'ec2'

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "elasticache"}, "parsed": false}
```

Copied from original issue: ansible/ansible-modules-core#2928

**Comment by ansibot** [comment]
From @ansibot on 2016-02-01T04:03:16Z



@jsdalton, @alachaum, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @willthames
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
@willthames Can we close this as this part of code has been refactored already ?

**Comment by ansibot** [comment]
cc @jillr @s-hertel @tremble
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
Closing this. As code is refactored. Please feel free to open a new issue in [`community.aws`](https://github.com/ansible-collections/community.aws) since this module is moved in this repo. 

Thanks.
